### DISCUSSION TITLE
How does refPath works on populating?

### DISCUSSION THREAD

**Discussion by tankotun** [Automattic/mongoose#13347] [Q&A]
Hello!

I am trying to retrieve documents using the lookup operator during aggregation in MongoDB. However, the "from" field in the lookup operator does not allow dynamic values such as in-schema variables. From what I have found, it must be a string literal. However, in Mongoose, we can create dynamic references using the "refPath" field.

My question is, how does the mechanism behind this dynamic referencing work in Mongoose? Is there any way to implement the same approach in the MongoDB aggregation pipeline?

Thank you for your help.

**Comment by vkarpov15** [comment]
Mongoose's `populate()` function doesn't use `$lookup` at all. That's why `refPath` can pull the referenced model name from the subdocument: Mongoose populate creates a separate `find()` query for each path you're trying to `populate()`. Can't do that with `$lookup` as far as I know.

**Reply by hasezoey** [reply]
`$lookup` (aggregation)'s `from` must be a string literal and does not support being dynamic, see [this mongodb question](https://www.mongodb.com/community/forums/t/dynamic-from-in-lookup/1596) and the [tracking issue SERVER-22497](https://jira.mongodb.org/browse/SERVER-22497)
